src/once_cell.rs

```diff
@@ -101,3 +110,76 @@ impl<T> GILOnceCell<T> {
         Ok(())
     }
 }
+
+/// Interns `text` as a Python string and stores a reference to it in static storage.
```

Comment:
Also, it should be elaborated how this differs from `PyString::intern`.

Reply:
> "Intern" or "Interns"?

Looking at other doc comments, they are all over the place w.r.t. this. Personally, I have no substantiated opinion and would be glad to change this if somebody has.

> Also, it should be elaborated how this differs from PyString::intern.

Isn't this the "and stores a reference to it in static storage." part?

Reply:
I would get more explicit:

* `PyString::intern` uses CPython's interning which is shared everywhere in the interpreter
* While the strings created here are only usable and available from Rust
* Two `intern!("foo")` create two independent strings

The last point makes me wonder, should this use `PyString::intern` instead of `PyString::new`? Then they would be much less different and more efficient.

Reply:
> The last point makes me wonder, should this use PyString::intern instead of PyString::new?

I just started to type that this does use `PyString::intern` but than noticed that I accidentally committed usage of `PyString::new` when adding `isolate_from_dyn_env`. Sorry for that and indeed this should definitely use `PyString::intern` to achieve the aimed at performance similar to using `_Py_Identifier`.

So after fixing that last commit back to `PyString::intern`, do you think the "static storage" part is sufficient?

Reply:
(c.f. https://github.com/PyO3/pyo3/pull/2269#issuecomment-1086928064 for the performance of using `PyString::intern`.)

Reply:
Yes, thanks!